### PR TITLE
added icon in form validation in #22762

### PR DESCRIPTION
**Pull request by chiraggmodi**
added form validation icon mention in #22762

### CODE REVIEW

**Comment by chiraggmodi** [comment]
can any one let me know what is the right way for this **Avoid qualifying class selectors with an element.** validation issue?

File: scss/mixins/_forms.scss [hunk 1]
```diff
@@ -40,6 +52,13 @@
     }
   }
 
+  // Validation icon top for textarea
+  textarea.form-control,
```
Comment (chiraggmodi) [inline scss/mixins/_forms.scss]: can any one please guide me on this issue?

**Comment by XhmikosR** [comment]
Doesn't this work?

```scss
element {
  .class {
  }
}
```

File: scss/mixins/_forms.scss (hunk 1 above)
Comment (XhmikosR) [inline scss/mixins/_forms.scss]: Doesn't this work?

```scss
element {
  &.class {
  }
}
```

EDIT: nah, it would still complain.

Comment (XhmikosR) [inline scss/mixins/_forms.scss]: See https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#qualifyingelement

Comment (chiraggmodi) [inline scss/mixins/_forms.scss]: @XhmikosR  Thank you!!

File: scss/mixins/_forms.scss [hunk 2]
```diff
@@ -10,24 +10,36 @@
...
+      padding-right: $input-btn-padding-x * 3;
+      background-repeat: no-repeat;
+      background-position: center right calc(#{$input-height-inner} / 4);
+      background-size: calc(#{$input-height-inner} / 2) calc(#{$input-height-inner} / 2);
+
+      @if $state == "valid" {
+        background-image: $form-icon-success;
+      }
+
+      @if $state == "invalid" {
```
Comment (vsn4ik) [inline scss/mixins/_forms.scss]: Maybe `@else if`...

Comment (XhmikosR) [inline scss/mixins/_forms.scss]: Yup definitely good suggestion @vsn4ik

Comment (chiraggmodi) [inline scss/mixins/_forms.scss]: @vsn4ik and @XhmikosR made changes as per suggestion.

**Comment by XhmikosR** [comment]
@mdo: what do we do with this? I think we better drop this from a quick look...

**Comment by mdo** [comment]
New PR at #26824.

**Pull request closed without merging**